src/mon/PGMap.cc

```diff
@@ -861,27 +876,46 @@ void PGMapDigest::dump_cluster_stats(stringstream *ss,
     f->dump_int("total_used_bytes", osd_sum.kb_used * 1024ull);
     f->dump_int("total_avail_bytes", osd_sum.kb_avail * 1024ull);
     f->dump_float("total_percent_used", used * 100);
-    if (verbose) {
-      f->dump_int("total_objects", pg_sum.stats.sum.num_objects);
+    f->close_section();
+    f->open_object_section("stats_by_class");
+    for (auto& i : osd_sum_by_class) {
+      f->open_object_section(i.first.c_str());
+      f->dump_int("total_bytes", i.second.kb * 1024ull);
+      f->dump_int("total_used_bytes", i.second.kb_used * 1024ull);
+      f->dump_int("total_avail_bytes", i.second.kb_avail * 1024ull);
+      float used = 0.0;
+      if (i.second.kb > 0) {
+	used = ((float)i.second.kb_used / i.second.kb);
+      }
+      f->dump_float("total_percent_used", used * 100);
+      f->close_section();
     }
     f->close_section();
   } else {
     ceph_assert(ss != nullptr);
     TextTable tbl;
+    tbl.define_column("CLASS", TextTable::LEFT, TextTable::LEFT);
     tbl.define_column("SIZE", TextTable::LEFT, TextTable::RIGHT);
     tbl.define_column("AVAIL", TextTable::LEFT, TextTable::RIGHT);
     tbl.define_column("USED", TextTable::LEFT, TextTable::RIGHT);
     tbl.define_column("%USED", TextTable::LEFT, TextTable::RIGHT);
-    if (verbose) {
-      tbl.define_column("OBJECTS", TextTable::LEFT, TextTable::RIGHT);
+    for (auto& i : osd_sum_by_class) {
+      tbl << i.first;
+      tbl << stringify(byte_u_t(i.second.kb*1024))
+	  << stringify(byte_u_t(i.second.kb_avail*1024))
+	  << stringify(byte_u_t(i.second.kb_used*1024));
+      float used = 0.0;
+      if (i.second.kb > 0) {
+	used = ((float)i.second.kb_used / i.second.kb);
+      }
+      tbl << percentify(used*100);
+      tbl << TextTable::endrow;
     }
+    tbl << "TOTAL";
```

Comment:
might want to updated https://github.com/ceph/ceph/blob/master/doc/rados/operations/monitoring.rst accordingly. search for `OBJECTS` in it.